test/e2e/instrumentation/monitoring/custom_metrics_stackdriver.go

```diff
@@ -176,13 +176,13 @@ func verifyResponsesFromCustomMetricsAPI(f *framework.Framework, customMetricsCl
 		framework.Failf("Failed to retrieve a list of supported metrics: %s", err)
 	}
 	gotCustomMetric, gotUnusedMetric := false, false
+	// Adapter will return any metrics associated with Kubernetes monitored resources. Verify that
+	// metrics created in this test are returned.
 	for _, resource := range resources.APIResources {
-		if resource.Name == "*/"+CustomMetricName {
+		if resource.Name == "*/custom.googleapis.com|"+CustomMetricName {
```

Comment:
What exactly are you trying to test here? Since this is a test, I'd prefer to see an explicit list of things to verify, rather than  a loop with cryptic strings.

Reply:
I'm trying (and actually it works :)) to verify that the list of discovered Stackdriver metrics contains a custom metrics that were created by this test. Custom Metrics - SD Adapter used to support only metrics with "custom.googleapis.com" prefix which wasn't specified explicitly. After PR https://github.com/GoogleCloudPlatform/k8s-stackdriver/pull/155, we support metrics with any prefix, i.e. "container.googleapis.com/container/cpu/usage_time".

The discovery now returns also a list of system metrics (i.e. "container.googleapis.com/*) and specifies full names of custom metrics (although the names with prefix "custom.googleapis.com" skipped will still work). Therefore this test needs to change slightly.

Changed it to be more explicit about what is tested.